tests/test_AIOConsumer.py

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+import pytest
+import asyncio
+import concurrent.futures
+from unittest.mock import Mock, patch, AsyncMock
+
+from confluent_kafka import TopicPartition
+from confluent_kafka.aio._AIOConsumer import AIOConsumer
+
+
+class TestAIOConsumer:
+    """Unit tests for AIOConsumer class - focusing on async wrapper logic."""
+
+    @pytest.fixture
+    def mock_consumer(self):
+        """Mock the underlying confluent_kafka.Consumer."""
+        with patch('confluent_kafka.aio._AIOConsumer.confluent_kafka.Consumer') as mock:
+            yield mock
+
+    @pytest.fixture
+    def mock_common(self):
+        """Mock the _common module callback wrapping."""
+        with patch('confluent_kafka.aio._AIOConsumer._common') as mock:
+            yield mock
+
+    @pytest.fixture
+    def basic_config(self):
+        """Basic consumer configuration."""
+        return {
+            'bootstrap.servers': 'localhost:9092',
+            'group.id': 'test-group',
+            'auto.offset.reset': 'earliest'
+        }
+
+    @pytest.mark.asyncio
+    async def test_constructor_behavior(self, mock_consumer, mock_common, basic_config):
+        """Test constructor creates consumer with correct configuration."""
```

Comment:
Not quite sure what we're testing for -- that _max_workers on executor isn't overwritten by consumer max_workers?